optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/ExperimentalConstraintCollectors.java

```diff
@@ -158,7 +166,7 @@
      * @param <C> type of the fact endpoints
      * @return never null
      */
-    public static <A, C extends Comparable<C>> UniConstraintCollector<A, IntervalTree<A, C>, ConsecutiveIntervalData<A, C>>
+    public static <A, C extends Comparable<C>> UniConstraintCollector<A, IntervalTree<A, C>, ConsecutiveIntervalDataImpl<A, C>>
             consecutiveIntervals(Function<A, C> startMap, Function<A, C> endMap) {
```

Comment:
Returning `Impl`s defeats the point of separating the API. Cast the return if you need to.

Reply:
The interval data types I'll do later; they are unused in the points consecutive collector